File: src/main/java/im/joker/device/Device.java
package im.joker.device;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Objects;

@AllArgsConstructor
@NoArgsConstructor
@Builder
@Data
public class Device implements IDevice {

    private String deviceId;

    private String accessToken;

    private String username;

    private String name;

    private String userId;

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        Device device = (Device) o;
        return Objects.equals(deviceId, device.deviceId) &&
                Objects.equals(username, device.username);
    }

    @Override
    public int hashCode() {
        return Objects.hash(deviceId, username);
    }

    @Override
    public String getUserId() {
        return userId;
    }
}

File: src/main/kotlin/im/joker/device/Device.kt
package im.joker.device

/**
 * @Author: mkCen
 * @Date: 2020/11/21
 * @Time: 18:20
 * @Desc:
 */
data class Device(val deviceId: String,
                  val accessToken: String,
                  val username: String,
                  val name: String?,
                  val deviceAvatar: String?,
                  val userId: String) {


}